### PR TITLE
Add line number interval setting

### PR DESCRIPTION
**Pull request by djweber**
This adds a setting to control the interval at which line numbers are rendered in the editor, described in #36981. You can specify a numerical interval as well as whether or not to show the line number for the current cursor position.

screen shot 2017-10-30 at 3 12 49 pm

screen shot 2017-10-30 at 3 14 59 pm

### CODE REVIEW

**Comment by wiktor-k** [comment]
Looks very good! Thanks.

**Review by RubenMateus — APPROVED** [review]

File: src/vs/editor/common/config/commonEditorConfig.ts [hunk 1]
```diff
@@ -207,6 +207,32 @@ const editorConfiguration: IConfigurationNode = {
 			'default': 'on',
 			'description': nls.localize('lineNumbers', "Controls the display of line numbers. Possible values are 'on', 'off', and 'relative'. 'relative' shows the line count from the current cursor position.")
 		},
+		'editor.lineNumberInterval': {
```
Comment (alexandrudima) [inline src/vs/editor/common/config/commonEditorConfig.ts]: IMHO, the introduction of the OR type makes this setting too difficult to configure. It is technically correct, but does not align with the other editor settings.

> I would remove the usage of the OR type.

**Review by alexdima — CHANGES_REQUESTED** [review]
For our users sake, who will have to figure out how to configure this setting, I would kindly ask that we remove the OR type for this setting... IMHO, it makes it too complicated.

**Comment by wiktor-k** [comment]
I've [fixed issues](https://github.com/wiktor-k/vscode/tree/line-number-interval) reported by @alexandrudima (splitting settings) on top of @DdWr's changes but as I can't push these changes to this PR.

Should I create a new PR for that or would it be possible for maintainers to push my changes to this PR?

Thanks in advance for help!

**Comment by DdWr** [comment]
@alexandrudima I thought it would be acceptable to use the union type since there are several throughout the configuration settings, 'editor.quickSuggestions' being a good example:

https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/config/commonEditorConfig.ts#L355

As requested, I `lineNumberInterval` is no longer a union type.

**Comment by alexandrudima** [comment]
The `editor.quickSuggestions` is a union type for backwards compatibility purposes and should not be used as a good example. It shipped for a couple of years as a boolean, and then someone in the team refined it.

**Review by alexdima — APPROVED** [review]

**Comment by alexandrudima** [comment]
@DdWr Thank you for your contribution! :heart:

**Comment by elektronik2k5** [comment]
I would love to have a setting to control the line interval, as suggested here: https://github.com/Microsoft/vscode/issues/36981#issuecomment-340044575. The default 10 lines is too sparse, but using a configurable interval, I could set it to 5 or 2 to have just enough line numbers.

**Comment by DdWr** [comment]
@elektronik2k5 The work I did on this PR did have a configurable interval length, but for whatever reason it was completely changed and hard-coded back to being an interval of 10. @alexandrudima can you please clarify why these changes were made?

**Comment by TrevorSayre** [comment]
@alexandrudima Is there any chance we can get back the ability to specify an interval?

**Comment by alexandrudima** [comment]
@dotweber @TrevorSayre  

It perhaps does not make sense, but I'm actively trying to add as little as possible to VS Code. And when something is added, I'm trying as much as possible to make it simple to use / configure / discover.

The initial feature request at #36981 had 1 upvote, so AFAIK there were a grand total of 3 interested people in it. The feature request does not mention a configurable interval, so I assumed @dotweber is overdelivering. Plus the screenshots appear to use the value 10. Plus I honestly believe 10 is a good default, as the line numbers are presented in base 10.

I am open to bring the possibility to configure the line numbers interval. But let's start with a fresh feature request (i.e. please create one); PR also most welcome.

**Comment by F1LT3R** [comment]
I would also like to see the interval configurable.
And it would be good to have an option for `current` only, eg:

```json
{
   "editor.lineNumbers": "current"
}
```

**Pull request merged**